File: 1894-merge-strings-alternately/merge-strings-alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {
    let p1 = 0;
    let p2 = 0;
    let mergedString = ""

    while (p1 < word1.length && p2 < word2.length) {
        mergedString += word1[p1] + word2[p2];
        p1++
        p2++
    }


    while (p2 < word2.length) {
        mergedString += word2[p2]
        p2++
    }

     while (p1 < word1.length) {
        mergedString += word1[p1]
        p1++
    }

    return mergedString;
};     

File: 1894-merge-strings-alternately/merge-strings-alternately.go
func mergeAlternately(word1 string, word2 string) string {
    n1, n2 := len(word1), len(word2)
    i, j := 0,0
    var result []byte
   
    for i < n1 || j < n2 {
        if i < n1 {
            result = append(result, word1[i])
            i++
        }
        if j < n2 {
            result = append(result, word2[j])
            j++
        }
    }

    return string(result);
}